src/jaxsim/math/joint_model.py

```diff
@@ -226,59 +221,54 @@ def successor_H_child(self, joint_index: jtp.IntLike) -> jtp.Matrix:
         return self.suc_H_i[joint_index]
 
 
-@functools.partial(jax.jit, static_argnames=["joint_type"])
+@jax.jit
 def supported_joint_motion(
-    joint_type: JointType | JointDescriptor, joint_position: jtp.VectorLike
+    joint_type: JointType, joint_axis: JointGenericAxis, joint_position: jtp.VectorLike
```

Comment:
This method is not part of the public APIs and users should not use it directly. However, I want to keep the possibility to support other type of joints that might require different metadata beyond or instead of the current `joint_axis`. Would you mind moving it as last function argument, also enforcing not to be a keyword? I mean the following: 

```suggestion
    joint_type: JointType, joint_position: jtp.VectorLike, joint_axis: JointGenericAxis, /
```

The forward slash positioned at the end should prevent passing any of the argument as keyword, therefore maintaining this method compatible with vmap. In the future, the signature can become:

```python
 joint_type: JointType, joint_position: jtp.VectorLike, *args
```

expecting args[0] to be `joint_axis` if `joint_type` is either Revolute or Prismatic, closed inside the `compute_{R,P}` functions.

Reply:
Btw this is also what I suggested in the original proposal :) https://github.com/ami-iit/jaxsim/issues/151#issuecomment-2112025292

Reply:
Sure, done in https://github.com/ami-iit/jaxsim/pull/153/commits/16edc053fb8d6aeed9da1f56c90f74faaa968443